Duplicaat BAG laadprocessen (downloads) van GDS2 worden niet goed afgehandeld

### Probleem
Het komt voor dat kadaster duplicaat downloads op gds2 zet voor bag (net als voor brk) we vangen dat niet elegant op. 
Zo'n laadproces zou als duplicaat moeten worden gemarkeerd en het ophalen moet doorlopen.

### Versie
versie info: <!-- ga naar http://<server:poort>/brmo-service/about.jsp en kopier/plak --> 1.5.2.x

### Server log
<details><summary>server log:</summary>

```

Gestart op: 2018-06-21 16:11:52, een paar seconden geleden.
Totaal: 2
Verwerkt: 0
Status: Uitvoeren SOAP request naar Kadaster afgerond
Proces afgebroken vanwege een exception:
nl.b3p.brmo.loader.util.BrmoException: Fout bij loaden basisregistratie gegevens
        at nl.b3p.brmo.loader.BrmoFramework.loadFromStream(BrmoFramework.java:390)
        at nl.b3p.brmo.service.scanner.GDS2OphalenProces.laadBagAfgifte(GDS2OphalenProces.java:510)
        at nl.b3p.brmo.service.scanner.GDS2OphalenProces.verwerkAfgiftes(GDS2OphalenProces.java:766)
        at nl.b3p.brmo.service.scanner.GDS2OphalenProces.execute(GDS2OphalenProces.java:386)
        at nl.b3p.brmo.service.stripes.GDS2OphalenUitvoerActionBean.execute(GDS2OphalenUitvoerActionBean.java:132)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at net.sourceforge.stripes.controller.DispatcherHelper$6.intercept(DispatcherHelper.java:456)
        at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:176)
        at net.sourceforge.stripes.controller.BeforeAfterMethodInterceptor.intercept(BeforeAfterMethodInterceptor.java:113)
        at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:173)
        at org.stripesstuff.plugin.waitpage.WaitPageInterceptor.intercept(WaitPageInterceptor.java:125)
        at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:173)
        at net.sourceforge.stripes.controller.ExecutionContext.wrap(ExecutionContext.java:86)
        at net.sourceforge.stripes.controller.DispatcherHelper.invokeEventHandler(DispatcherHelper.java:454)
        at net.sourceforge.stripes.controller.DispatcherServlet.invokeEventHandler(DispatcherServlet.java:278)
        at net.sourceforge.stripes.controller.DispatcherServlet.service(DispatcherServlet.java:160)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at net.sourceforge.stripes.controller.StripesFilter.doFilter(StripesFilter.java:260)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:614)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:617)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:518)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1091)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:668)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1527)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1484)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Unknown Source)
Caused by: nl.b3p.brmo.loader.util.BrmoDuplicaatLaadprocesException: Laadproces al gerund voor bestand BAG-Extract-DNLDLXAM02-1815-5015126-19062018-20062018.zip (c3af7874-18c0-4468-bc7e-02ce4b8fda67)/9999MUT19062018-20062018.zip/9999MUT19062018-20062018-000001.xml
        at nl.b3p.brmo.loader.StagingProxy.loadBr(StagingProxy.java:618)
        at nl.b3p.brmo.loader.BrmoFramework.loadFromStream(BrmoFramework.java:388)
        ... 43 more

```

</details>

--- a/brmo-loader/src/main/java/nl/b3p/brmo/loader/BrmoFramework.java
+++ b/brmo-loader/src/main/java/nl/b3p/brmo/loader/BrmoFramework.java
@@ -318,7 +318,9 @@ public class BrmoFramework {
     }
 
     /**
-     * NB na gebruik zelf de database verbinding sluiten / opruimen met {@link #closeBrmoFramework()}
+     * laden van BR data uit een bestand. <br>
+     * NB na gebruik zelf de database verbinding sluiten / opruimen met
+     * {@link #closeBrmoFramework()}.
      *
      * @param type basis registratie type
      * @param fileName bestand
@@ -383,15 +385,35 @@ public class BrmoFramework {
         }
     }
 
-    public void loadFromStream(String type, InputStream stream, String fileName) throws BrmoException {
+    /**
+     * laden van BR data uit een stream.
+     *
+     * @param type type registratie, bijv. {@value BrmoFramework#BR_BRK}
+     * @param stream datastream
+     * @param fileName te gebruiken bestandsnaam om laadproces te identificeren
+     * @throws BrmoException als er een algemene fout optreed
+     * @throws BrmoDuplicaatLaadprocesException als het "bestand"
+     * {@code fileName} al geladen is
+     * @throws BrmoLeegBestandException als het "bestand" {@code fileName} leeg
+     * is
+     */
+    public void loadFromStream(String type, InputStream stream, String fileName)
+            throws BrmoException, BrmoDuplicaatLaadprocesException, BrmoLeegBestandException {
         try {
             stagingProxy.loadBr(stream, type, fileName, null);
-        } catch(Exception e) {
+        } catch (Exception e) {
+            if (e instanceof BrmoDuplicaatLaadprocesException) {
+                throw (BrmoDuplicaatLaadprocesException) e;
+            }
+            if (e instanceof BrmoLeegBestandException) {
+                throw (BrmoLeegBestandException) e;
+            }
             throw new BrmoException("Fout bij loaden basisregistratie gegevens", e);
         }
     }
 
     /**
+     * laden van BR data uit een stream.
      *
      * @param type type registratie, bijv. {@value BrmoFramework#BR_BRK}
      * @param stream datastream

--- a/brmo-service/src/main/java/nl/b3p/brmo/service/scanner/GDS2OphalenProces.java
+++ b/brmo-service/src/main/java/nl/b3p/brmo/service/scanner/GDS2OphalenProces.java
@@ -53,6 +53,8 @@ import javax.xml.ws.BindingProvider;
 import javax.xml.ws.handler.Handler;
 import nl.b3p.brmo.loader.BrmoFramework;
 import nl.b3p.brmo.loader.entity.BrkBericht;
+import nl.b3p.brmo.loader.util.BrmoDuplicaatLaadprocesException;
+import nl.b3p.brmo.loader.util.BrmoLeegBestandException;
 import nl.b3p.brmo.loader.xml.BrkSnapshotXMLReader;
 import nl.b3p.brmo.persistence.staging.AutomatischProces;
 import static nl.b3p.brmo.persistence.staging.AutomatischProces.LOG_NEWLINE;
@@ -506,7 +508,22 @@ public class GDS2OphalenProces extends AbstractExecutableProces {
                         l.updateStatus(msg);
                         l.addLog(msg);
                         log.debug(msg);
-                        brmo.loadFromStream("bag", new CloseShieldInputStream(innerzip), getLaadprocesBestandsnaam(a) + "/" + entry.getName() + "/" + innerentry.getName());
+                        try {
+                            brmo.loadFromStream(BrmoFramework.BR_BAG,
+                                    new CloseShieldInputStream(innerzip),
+                                    getLaadprocesBestandsnaam(a) + "/" + entry.getName() + "/" + innerentry.getName()
+                            );
+                        } catch (BrmoDuplicaatLaadprocesException d) {
+                            msg = "Duplicaat laadproces. " + d.getLocalizedMessage();
+                            l.updateStatus(msg);
+                            l.addLog(msg);
+                            log.warn(msg);
+                        } catch (BrmoLeegBestandException e) {
+                            msg = "Leeg bestand voor laadproces. " + e.getLocalizedMessage();
+                            l.updateStatus(msg);
+                            l.addLog(msg);
+                            log.info(msg);
+                        }
                         innerentry = innerzip.getNextEntry();
                     }
                 } else {

